### PR TITLE
[CARBONDATA-4009] Fix PartialQueries not hitting mv

### PR DESCRIPTION
**Pull request by Indhumathi27**
### Why is this PR needed?

If a MV having aggregation, is created with all columns present in main table, then some partial-queries are not hitting mv. 

 ### What changes were proposed in this PR?
1. When all columns are given in projection in a aggregate query, logical plan will not have Project node. Added Project node to logical plan in this case, to match partialqueries also to mv.
2. In GroupByGroupByNochildData pattern, added check for checking CAST(aggregate expression) and used group by result to get final compensation result
    
 ### Does this PR introduce any user interface change?
 - No

 ### Is any new testcase added?
 - Yes

### CODE REVIEW

**Comment by CarbonDataQA1** [comment]
Build Success with Spark 2.4.5, Please check CI http://121.244.95.60:12545/job/ApacheCarbon_PR_Builder_2.4.5/2462/

**Comment by CarbonDataQA1** [comment]
Build Success with Spark 2.3.4, Please check CI http://121.244.95.60:12545/job/ApacheCarbonPRBuilder2.3/4205/

File: mv/plan/src/main/scala/org/apache/carbondata/mv/plans/modular/Modularizer.scala [hunk 1]
```diff
@@ -66,6 +68,22 @@ abstract class Modularizer[TreeType <: TreeNode[TreeType]] {
     makeupAliasMappings(mplans)
   }
 
+  private def transformPlan(plan: LogicalPlan) = {
```
Comment (akashrn5) [inline mv/plan/src/main/scala/org/apache/carbondata/mv/plans/modular/Modularizer.scala:71]: please change the method name to something meaningful according to functionality

**Comment by CarbonDataQA1** [comment]
Build Failed  with Spark 2.3.4, Please check CI http://121.244.95.60:12545/job/ApacheCarbonPRBuilder2.3/4211/

**Comment by Indhumathi27** [comment]
retest this please

**Comment by CarbonDataQA1** [comment]
Build Success with Spark 2.4.5, Please check CI http://121.244.95.60:12545/job/ApacheCarbon_PR_Builder_2.4.5/2468/

**Comment by CarbonDataQA1** [comment]
Build Success with Spark 2.4.5, Please check CI http://121.244.95.60:12545/job/ApacheCarbon_PR_Builder_2.4.5/2472/

**Comment by CarbonDataQA1** [comment]
Build Success with Spark 2.3.4, Please check CI http://121.244.95.60:12545/job/ApacheCarbonPRBuilder2.3/4215/

**Comment by akashrn5** [comment]
LGTM

**Pull request closed without merging**